src/utils/ResolveDynamicValue.js

```diff
@@ -0,0 +1,27 @@
+import { RunnersUtils } from './RunnersUtils';
+
+export function resolveDynamicValue(key, context) {
+  const { currentScenarioData, visibleScenarios } = context;
+  const lastRunId = RunnersUtils.getLastRunId(currentScenarioData);
+
+  const dynamic = {
+    id: () => currentScenarioData?.id,
+    lastRunId: () => lastRunId,
+    csmSimulationRun: () => currentScenarioData?.lastRun?.csmSimulationRun,
+    state: () => currentScenarioData?.state,
+    name: () => currentScenarioData?.name,
+    masterId: () => currentScenarioData?.masterId,
+    parentId: () => currentScenarioData?.parentId,
+    ownerId: () => currentScenarioData?.ownerId,
+    solutionId: () => currentScenarioData?.solutionId,
+
+    visibleScenariosIds: () => visibleScenarios?.map((s) => s.id),
+
+    visibleScenariosSimulationRunsIds: () => visibleScenarios?.map((s) => s.lastRunId),
+
+    visibleScenariosCsmSimulationRunsIds: () => visibleScenarios?.map((s) => s.lastRun?.csmSimulationRun),
```

Comment:
I think there are some things to change here, because some of these fields do not exist in the scenario (Runner) object in the API v5 (see docs [here](https://github.com/Cosmo-Tech/cosmotech-api/blob/5.0.0-rc1/doc/Models/Runner.md)):
- the options `csmSimulationRun` and `visibleScenariosCsmSimulationRunsIds` can be removed, because the back-end no longer provides this value
- the "state" value should be read from `currentScenarioData?.lastRunInfo?.lastRunStatus`
- the "masterId" value should be read from `currentScenarioData?.rootId`
- the "ownerId" value should be read from `currentScenarioData?.createInfo?.userId`
- the value for "visibleScenariosSimulationRunsIds" should call the `RunnersUtils.getLastRunId` function